Valeu @antoniospneto @renatonlima @rvalyi pela revisão, 

Pela resposta do Antonio a implementação atende o caso do l10n_br_ie_search

Sobre a questão do Raphael se concordarem posso ver de incluir o campo, mas talvez seja mais código do que o necessário porque de alguma forma o programa vai precisar dar uma Mensagem de Erro com a orientação para usuário sobre o certificado não encontrado, essa mensagem precisa aparecer no momento da tentativa de uso desse Certificado, porque não é um campo requerido no cadastro da Empresa, então não sei dizer se colocando como um compute isso vai funcionar dessa forma esperada ( o método do compute vai retorna a mensagem de erro quando o campo for chamado e estiver vazio? Não lembro de ter feito algo nesse sentido, mas talvez seja possível e funcione ), acredito que quanto menos código melhor nesse caso.

Vou ver de melhorar essa mensagem de erro porque hoje esta muito simples "Certificate not found" para algo como "Certificate not found, you need to inform your E-CNPJ or E-NFe certificate in the Company" sugestões são bem vindas

O erro dos testes não tem relação com esse PR por isso para ficar verde está dependendo do https://github.com/OCA/l10n-brazil/pull/2894

https://github.com/OCA/l10n-brazil/actions/runs/7647673451/job/20839031978?pr=2892#step:8:2096

2024-01-24 23:52:29,896 563 ERROR odoo odoo.addons.l10n_br_cnpj_search.tests.test_receitaws: FAIL: TestReceitaWS.test_receita_ws_success
Traceback (most recent call last):
File "/__w/l10n-brazil/l10n-brazil/l10n_br_cnpj_search/tests/test_receitaws.py", line 38, in test_receita_ws_success
self.assertEqual(kilian.cnae_main_id.code, "4751-2/01")
AssertionError: False != '4751-2/01'

--- a/l10n_br_fiscal_certificate/models/res_company.py
+++ b/l10n_br_fiscal_certificate/models/res_company.py
@@ -3,7 +3,8 @@
 # License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
 
 
-from odoo import fields, models
+from odoo import _, fields, models
+from odoo.exceptions import ValidationError
 
 
 class ResCompany(models.Model):
@@ -20,3 +21,26 @@ class ResCompany(models.Model):
         string="NFe",
         domain="[('type', '=', 'nf-e')]",
     )
+
+    certificate = fields.Many2one(
+        comodel_name="l10n_br_fiscal.certificate",
+        compute="_compute_certificate",
+    )
+
+    def _compute_certificate(self):
+        for record in self:
+            certificate = False
+            if record.sudo().certificate_nfe_id:
+                certificate = record.sudo().certificate_nfe_id
+            elif record.sudo().certificate_ecnpj_id:
+                certificate = record.sudo().certificate_ecnpj_id
+
+            if not certificate:
+                raise ValidationError(
+                    _(
+                        "Certificate not found, you need to inform your e-CNPJ"
+                        " or e-NFe certificate in the Company."
+                    )
+                )
+
+            record.certificate = certificate

--- a/l10n_br_fiscal_certificate/tests/__init__.py
+++ b/l10n_br_fiscal_certificate/tests/__init__.py
@@ -1,0 +1,1 @@
+from . import test_certificate

--- a/l10n_br_fiscal_certificate/tests/test_certificate.py
+++ b/l10n_br_fiscal_certificate/tests/test_certificate.py
@@ -114,3 +114,20 @@ class TestCertificate(SavepointCase):
                     "file": self.certificate_invalid,
                 }
             )
+
+    def test_compute_field_to_get_certificate(self):
+        """Test compute field to get Certificate or e-CNPJ or e-NFe"""
+        company = self.env.company
+        with self.assertRaises(ValidationError):
+            assert company.certificate
+        cert = self.certificate_model.create(
+            {
+                "type": "nf-e",
+                "subtype": "a1",
+                "password": self.cert_passwd,
+                "file": self.certificate_valid,
+            }
+        )
+
+        company.certificate_nfe_id = cert
+        assert company.certificate

--- a/l10n_br_nfe/models/document.py
+++ b/l10n_br_nfe/models/document.py
@@ -864,16 +864,10 @@ class NFe(spec_models.StackedModel):
         return edocs
 
     def _processador(self):
-        certificate = False
-        if self.company_id.sudo().certificate_nfe_id:
-            certificate = self.company_id.sudo().certificate_nfe_id
-        elif self.company_id.sudo().certificate_ecnpj_id:
-            certificate = self.company_id.sudo().certificate_ecnpj_id
 
-        if not certificate:
-            raise UserError(_("Certificado não encontrado"))
         self._check_nfe_environment()
 
+        certificate = self.env.company.certificate
         certificado = cert.Certificado(
             arquivo=certificate.file,
             senha=certificate.password,

--- a/l10n_br_nfe/models/invalidate_number.py
+++ b/l10n_br_nfe/models/invalidate_number.py
@@ -9,8 +9,7 @@ from erpbrasil.transmissao import TransmissaoSOAP
 from nfelib.nfe.ws.edoc_legacy import NFCeAdapter as edoc_nfce, NFeAdapter as edoc_nfe
 from requests import Session
 
-from odoo import _, fields, models
-from odoo.exceptions import UserError
+from odoo import fields, models
 
 from odoo.addons.l10n_br_fiscal.constants.fiscal import EVENT_ENV_HML, EVENT_ENV_PROD
 
@@ -19,12 +18,10 @@ class InvalidateNumber(models.Model):
     _inherit = "l10n_br_fiscal.invalidate.number"
 
     def _processador(self):
-        if not self.company_id.sudo().certificate_nfe_id:
-            raise UserError(_("Certificado não encontrado"))
-
+        certificate = self.env.company.certificate
         certificado = cert.Certificado(
-            arquivo=self.company_id.sudo().certificate_nfe_id.file,
-            senha=self.company_id.sudo().certificate_nfe_id.password,
+            arquivo=certificate.file,
+            senha=certificate.password,
         )
         session = Session()
         session.verify = False

